### PR TITLE
Chore/items table and updates 14

### PR DESCRIPTION
**Pull request by SandrineCipolla**
- Add table "items" in BDD

- updated code to have StockDetails and StockItems (quantities are handled in StockItems)
- New route to display StockItems
- New component StockDetailsWithItems to display all informations

### CODE REVIEW

File: backend/src/controllers/stockController.ts [hunk 1]
```diff
@@ -91,20 +90,49 @@ export const getStockDetails = async (
...
+        if (res && res.json) {
+            res.json(items);
+        } else {
+            throw new Error(
+                "Response or res.json is undefined. Cannot call res.status and res.json for error handling."
+            );
+        }
+    } catch (err: any) {
+        console.error(err);
+        if (res && res.json) {
```
Comment (macreiben-dev) [inline backend/src/controllers/stockController.ts:114]: suggestion: affiner les message d'erreur du coup ?

File: backend/src/routes/stockRoutes.ts [hunk 2]
```diff
@@ -34,12 +33,26 @@ const configureStockRoutes = (): Router => {
...
 
+    //Route pour récupérer les items d'un stock via l'ID
+    router.get("/stocks/:ID/items", async (req, res) => {
+        const ID = Number(req.params.ID);
+        try {
+            const connection = await connectToDatabase();
+            await stockController.getStockItems(req, res, connection, ID);
+            connection.release();
+        } catch (error) {
+            //TODO :affiner les message d'erreur ( ex: ajouter la route et le verbe utilisés pour faciliter le debug)
```
Comment (macreiben-dev) [inline backend/src/routes/stockRoutes.ts:44]: suggestion: affiner le message d'erreur.

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -1,27 +1,31 @@
...
+        <Router>
+            <Header/>
+            <main>
+                <Routes>
+                    <Route path="/" element={<Navigate to="/home"/>}/>
+                    <Route path="/home" element={<Home/>}/>
+                    <Route path="/stocks" element={<StocksList/>}/>
+                    <Route path="/stocks/:ID" element={<StockDetailsWithItems/>}/>
+                    //TODO: check the need/relevance of these routes
+                    {/*<Route path="/stocks/:ID" element={<StockDetails />} />*/}
```
Comment (macreiben-dev) [inline frontend/src/App.tsx:22]: issue(blocking): commented code.

File: frontend/src/App.tsx [hunk 4]
```diff
@@ -1,27 +1,31 @@
...
+                    {/*<Route path="/stocks/:ID/items" element={<StockItems  ID={ID}/>} />*/}
```
Comment (macreiben-dev) [inline frontend/src/App.tsx:23]: issue(blocking): commented code to remove.

File: frontend/src/utils/StockAPIClient.ts [hunk 5]
```diff
@@ -46,9 +46,27 @@ export const fetchStockDetails = async (numericID: number): Promise<StockDetail>
     }
 };
 
-export const updateStockQuantity = async (numericID: number, quantity: number) => {
+export const fetchStockItems = async (numericID: number): Promise<StockItem[]> => {
+    const response = await fetch(`${apiUrl}/stocks/${numericID}/items`, getConfig);
```
Comment (macreiben-dev) [inline frontend/src/utils/StockAPIClient.ts:50]: question: c'est pas server url ?

Comment (SandrineCipolla) [reply frontend/src/utils/StockAPIClient.ts:50]: const apiUrl = ConfigManager.getApiServerUrl();
=>static getApiServerUrl() {
        return import.meta.env.VITE_API_SERVER_URL || 'http://localhost:3000/api/v1';
    }

File: backend/src/controllers/stockController.ts (hunk 1 above)
Comment (SandrineCipolla) [reply backend/src/controllers/stockController.ts:114]: cfr TODO

File: backend/src/routes/stockRoutes.ts (hunk 2 above)
Comment (SandrineCipolla) [reply backend/src/routes/stockRoutes.ts:44]: cfr TODO

File: frontend/src/App.tsx (hunk 4 above)
Comment (SandrineCipolla) [reply frontend/src/App.tsx:23]: cfr TODO

**Pull request merged**